agent/exploits/cve_2024_27198.py

```diff
@@ -0,0 +1,26 @@
+"""Agent Asteroid implementation for CVE-2024-27198"""
+
+import re
+
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "JetBrains TeamCity before 2023.11.4 authentication bypass allowing to perform admin actions"
+VULNERABILITY_REFERENCE = "CVE-2024-27198"
+VULNERABILITY_DESCRIPTION = (
+    "an authentication bypass vulnerability in the web component of TeamCity that arises from an alternative path "
+    "issue (CWE-288) and has a CVSS base score of 9.8 (Critical)"
+)
+RISK_RATING = "CRITICAL"
+
+
+class CVE202427198Exploit(webexploit.WebExploit):
+    def __init__(self) -> None:
+        super().__init__()
+        self.accept_pattern = re.compile(r"TeamCity")
+        self.match_pattern = re.compile(r'<user username=".*" .*\/>')
+        self.metadata = {
+            "title": VULNERABILITY_TITLE,
+            "description": VULNERABILITY_DESCRIPTION,
+            "reference": VULNERABILITY_REFERENCE,
+            "risk_rating": RISK_RATING,
+        }
```

Comment:
This pattern will look nicer :).

```suggestion
class CVE202427198Exploit(webexploit.WebExploit):
        accept_pattern = re.compile(r"TeamCity")
        match_pattern = re.compile(r'<user username=".*" .*\/>')
        metadata = {
            "title": VULNERABILITY_TITLE,
            "description": VULNERABILITY_DESCRIPTION,
            "reference": VULNERABILITY_REFERENCE,
            "risk_rating": RISK_RATING,
        }
```

Reply:
Agree.